src/Acuminator/Acuminator.Analyzers/StaticAnalysis/PXOverrideMismatch/PXOverrideMismatchAnalyzer.cs

```diff
@@ -0,0 +1,230 @@
+
+namespace Acuminator.Analyzers.StaticAnalysis.PXOverrideMismatch
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class PXOverrideMismatchAnalyzer : SymbolAnalyzersAggregator<IPXGraphAnalyzer>
+	{
+		public PXOverrideMismatchAnalyzer(CodeAnalysisSettings settings, params IPXGraphAnalyzer[] innerAnalyzers) : base(settings, innerAnalyzers)
+		{
+		}
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+			ImmutableArray.Create(Descriptors.PX1096_PXOverrideMustMatchSignature, Descriptors.PX1096_PXOverrideContainerIsNotPXGraphExtension);
+
+		protected override SymbolKind SymbolKind => SymbolKind.Method;
+
+		protected override void AnalyzeSymbol(SymbolAnalysisContext context, PXContext pxContext)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			if (context.Symbol is IMethodSymbol methodSymbol)
+			{
+				if (!methodSymbol.IsStatic && methodSymbol.HasAttribute(pxContext.AttributeTypes.PXOverrideAttribute, false))
+				{
+					AnalyzeMethod(context, pxContext, methodSymbol);
+				}
+			}
+		}
+
+		private void AnalyzeMethod(SymbolAnalysisContext context, PXContext pxContext, IMethodSymbol methodSymbol)
+		{
+			// Here we know that the method is not static and has the correct attribute.
+
+			var extensionType = GetPXGraphExtension(pxContext.PXGraphExtensionTypes, methodSymbol.ContainingType);
+			var graphType = FindFirstTypeArgumentOfPxGraphExtension(extensionType);
+
+			if (graphType != null)
+			{
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				var allTypes = GetAllBaseTypes(pxContext.PXGraphExtensionTypes, extensionType, graphType);
+
+				context.CancellationToken.ThrowIfCancellationRequested();
+
+				var hasMatchingMethod = allTypes
+					.SelectMany(t => t.GetMembers().OfType<IMethodSymbol>())
+					.Any(m => IsCompatibleMethod(m, methodSymbol));
+
+				if (!hasMatchingMethod)
+				{
+					var diagnostic = Diagnostic.Create(
+						Descriptors.PX1096_PXOverrideMustMatchSignature,
+						methodSymbol.Locations.First());
+
+					context.ReportDiagnosticWithSuppressionCheck(diagnostic, pxContext.CodeAnalysisSettings);
+				}
+			}
+			else
+			{
+				// PXOverride can only be used in a PXGraphExtension child type.
+				var diagnostic = Diagnostic.Create(
+					Descriptors.PX1096_PXOverrideContainerIsNotPXGraphExtension,
+					methodSymbol.Locations.First());
+
+				context.ReportDiagnosticWithSuppressionCheck(diagnostic, pxContext.CodeAnalysisSettings);
+			}
+		}
+
+		private static HashSet<INamedTypeSymbol> GetAllBaseTypes(
+			List<INamedTypeSymbol> pxGraphExtensionTypes,
+			params INamedTypeSymbol[] types)
+		{
+			var allTypes = new HashSet<INamedTypeSymbol>(SymbolEqualityComparer.Instance);
+
+			foreach (var type in types)
+			{
+				GetBaseTypes(pxGraphExtensionTypes, type, allTypes);
+			}
+
+			return allTypes;
+		}
+		private static void GetBaseTypes(
+			List<INamedTypeSymbol> pxGraphExtensionTypes,
```

Comment:
please use meaningful names for every code element. Names like `hashset` do not tell anything to the reader about what it really is, and how it is used. I can see that it is a hashset just from the parameter type, so there is no need to duplicate this info.

Reply:
This variable is renamed now.